build/update-hostreferences.ps1

```diff
@@ -20,6 +20,76 @@ function WriteLog
     Write-Host $Message
 }
 
+function ValidateNugetListOutput
+{
+    param (
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [System.String]
+        $Output,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [System.String]
+        $Command
+    )
+
+    if ($Output -like "*No packages found*")
+    {
+        WriteLog "Failed to get the latest package information via: $Command" -Throw
+    }
+}
+
+function GetPackageInfo
+{
+    param (
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [System.String]
+        $Name,
+
+        [Parameter(Mandatory=$false)]
+        [System.String]
+        $MajorVersion
+    )
+
+    $result = $null
+
+    if (-not [string]::IsNullOrWhiteSpace($MajorVersion))
```

Comment:
It looks like we always call the function with $MajorVersion. Why do we need the fork?

Reply:
If not major version is specified, we call this command `NuGet list $Name -Source $SOURCE -PreRelease` which returns a single result with the latest package version. 

When a major version is specified, we call this command `NuGet list $Name -Source $SOURCE -PreRelease -AllVersions` which returns an ordered collection. After this we iterate through the collection to find the first result that matches the major version.

Reply:
I see, "https://raw.githubusercontent.com/Azure/azure-functions-integration-tests/dev/integrationTestsBuild/V3/HostBuild2.json" can have MajorVersion in major version